crates/apollo-compiler/tests/input_coercion.rs

```diff
@@ -0,0 +1,50 @@
+
+#[test]
+fn test_graphql_float_variable_coercion() {
+    // Small schema with a Float in the input object
+    let sdl = r#"
+      type Car { id: ID! kilometers: Float! }
+      input CarInput { kilometers: Float! }
+      type Query { getCarById(id: ID!): Car }
+      type Mutation { insertACar(car: CarInput!): Car!
+      }
+    "#;
+
+    let parsed_schema = Schema::parse_and_validate(sdl, "sdl").unwrap();
+
+    let executable_mutation = ExecutableDocument::parse_and_validate(
+        &parsed_schema,
+        "mutation MyCarInsertMutation($car: CarInput!){ insertACar(car:$car) { id kilometers } }",
+        "MyCarInsertMutation",
+    )
+        .unwrap();
+
+    let operation = executable_mutation
+        .operations
+        .get(Some("MyCarInsertMutation"))
+        .unwrap();
+
+    let kilometers_value = 3000;
+
+    // Provide an integer for a Float field
+    let input_variables = serde_json_bytes::json!({ "car": { "kilometers": kilometers_value } });
+    let map = match input_variables {
+        serde_json_bytes::Value::Object(m) => m,
+        _ => unreachable!(),
+    };
```

Comment:
Could I ask you to add another test to this file that shows how coercion would fail? 

Since we have changed the coercion code to now be `value.as_f64().is_some_and(f64::is_finite)`, there is a chance for a `None` to be returned from `as_f64`. You might have to have `kilometers_value` be defined as a string, otherwise the compiler won't let this compile. So potentially two variations of the test like this:

```rust
// Infinity; should fail coercion
let kilometers_value = "1e1000";
```
and another one:
```rust
// i128 MAX; should also fail coercion, since it's > i64 MAX
let kilometers_value = "170141183460469231731687303715884105727";
```